pkg/registry/common/memory/nse_server_test.go

```diff
@@ -177,6 +183,180 @@ func TestNetworkServiceEndpointRegistryServer_RegisterAndFindByLabelWatch(t *tes
 	require.True(t, proto.Equal(expected, <-ch))
 }
 
+func TestNetworkServiceEndpointRegistryServer_DataRace(t *testing.T) {
```

Comment:
This test looks invalid to me.
Because this test made a race in the test (line 222)
This can not be reproduced in real work if we'll use a real client.

Please consider reworking this test to catch a true race condition.